packages/renderer/src/lib/deployments/DeploymentDetails.spec.ts

```diff
@@ -46,106 +46,171 @@ const deployment: V1Deployment = {
     template: {},
   },
 };
-
-vi.mock('/@/stores/kubernetes-contexts-state', async () => {
+vi.mock(import('../kube/resources-listen'), async importOriginal => {
+  // we want to keep the original nonVerbose
+  const original = await importOriginal();
   return {
-    kubernetesCurrentContextDeployments: vi.fn(),
-    kubernetesCurrentContextEvents: vi.fn(),
+    ...original,
+    listenResources: vi.fn(),
+    isKubernetesExperimentalMode: vi.fn(),
   };
 });
 
-const showMessageBoxMock = vi.fn();
+vi.mock('/@/stores/kubernetes-contexts-state');
 
-beforeAll(() => {
-  Object.defineProperty(window, 'kubernetesDeleteDeployment', { value: kubernetesDeleteDeploymentMock });
-  Object.defineProperty(window, 'kubernetesReadNamespacedDeployment', { value: vi.fn() });
-  Object.defineProperty(window, 'showMessageBox', { value: showMessageBoxMock });
+beforeEach(() => {
+  vi.resetAllMocks();
+  router.goto('http://localhost:3000');
 });
 
-test('Expect redirect to previous page if deployment is deleted', async () => {
-  showMessageBoxMock.mockResolvedValue({ response: 0 });
+type initListsReturnType = {
+  updateDeployments: (objects: KubernetesObject[]) => void;
+  updateEvents: (objects: CoreV1Event[]) => void;
+};
 
-  const routerGotoSpy = vi.spyOn(router, 'goto');
+describe.each<{
+  experimental: boolean;
+  initLists: (deployments: KubernetesObject[], events: CoreV1Event[]) => initListsReturnType;
+}>([
+  {
+    experimental: false,
+    initLists: (deployments: KubernetesObject[], events: CoreV1Event[]): initListsReturnType => {
```

Comment:
I will factorize this code which is duplicated in ServiceDetails and NodeDetails tests in a specific PR